Allow extensions to be able to make requests to the typescript extension's tsserver via commands

orta opened the pull request:
This PR relates to https://github.com/microsoft/TypeScript-Website/issues/2156 and https://github.com/microsoft/vscode/pull/138211

This allows other extensions to make calls to the TypeScript Language Server, saving the need to create a second set of parallel tsservice-like infra in extensions like the typescript playground.

---

Tested by an extensions which calls:

```ts
console.log("Sending")
vscode.commands.executeCommand('typescript.tsserverRequest', 'emit-output', { file: "/Users/ortatherox/dev/js/berry/packages/plugin-git/sources/GitResolver.ts" } ).then(r => {
	console.log("Sent")
	console.log(r)
})
```

Which correctly returns the response to the request on the other side:

![Screen Shot 2021-12-01 at 4 37 37 PM](https://user-images.githubusercontent.com/49038/144276616-0b060bb5-3cad-4480-b461-ccad8d9f14a0.png)

orta commented:
( Fails look unrelated )

mjbvz commented on extensions/typescript-language-features/src/commands/tsserverRequests.ts:
```diff
@@ -0,0 +1,29 @@
+		// A cancellation token cannot be passed through the command infrastructure
+		const token = nulToken;
+		try {
+			return this.lazyClientHost.value.serviceClient.execute(requestID, args, token, config);
```
For the error handling, should we be awaiting the result of this call? Not sure when the code will throw as it is currently written

mjbvz reviewed (commented):
Implementation looks good. I'm a little concerned about exposing this though since it gives extensions low level control and could make it difficult to understand why the server is in an odd state

What specific commands do you need to run for the playground?

orta commented:
Updated based on a chat we had yesterday. 

I've removed my custom `try catch` error handling, at this point all the commands are reads and I don't think it should be possible to knock over the tsserver with them.

I don't know all of the commands the playground needs, but realistically it's probably a subset of the amount I've added here. It's enough to provide pretty solid introspection tooling.

mjbvz commented:
Thanks! LGTM
